parser-typechecker/src/Unison/Codebase/Serialization/V1.hs

```diff
@@ -709,6 +715,47 @@ getRawBranch =
     <*> getMap getNameSegment (RawHash <$> getHash)
     <*> getMap getNameSegment getHash
 
+-- `getBranchDependencies` consumes the same data as `getRawBranch`
+getBranchDependencies :: MonadGet m => m (BD.Branches n, BD.Dependencies)
+getBranchDependencies = do
+  (terms1, types1) <- getTermStarDependencies
+  (terms2, types2) <- getTypeStarDependencies
+  childHashes <- fmap (RawHash . snd) <$> getList (getPair skipText getHash)
+  editHashes <- Set.fromList . fmap snd <$> getList (getPair skipText getHash)
+  pure ( childHashes `zip` repeat Nothing
+       , BD.Dependencies editHashes (terms1 <> terms2) (types1 <> types2) )
+  where
+  -- returns things, metadata types, metadata values
+  getStarReferences ::
+    (MonadGet m, Ord r) => m r -> m ([r], [Metadata.Value])
+  getStarReferences getR = do
+    void $ getList getR -- throw away the `facts`
```

Comment:
You might want a `skipR` just like you have a `skipText`, and `skipList` combinator as well. This line is scanning every byte in the fact dimension only to throw it all away.